packages/@aws-cdk/aws-apigatewayv2/lib/http/api-mapping.ts

```diff
@@ -0,0 +1,87 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnApiMapping, CfnApiMappingProps } from '../apigatewayv2.generated';
+import { IApiMapping, IDomainName, IStage } from '../common';
+import { IHttpApi } from '../http/api';
+
+/**
+ * Represents a Route for an HTTP API.
+ */
+export interface IHttpApiMapping extends IApiMapping {
+  /**
+   * The HTTP API associated with this API mapping.
+   */
+  // readonly httpApi: IHttpApi;
+}
+
+/**
+ * Properties used to create the HttpApiMapping resource
+ */
+export interface HttpApiMappingProps {
+  /**
+   * Name for the API Mapping resource
+   * @default - id of the HttpApiMapping construct.
+   */
+  readonly apiMappingName?: string;
+  /**
+   * API for the HttpApiMapping resource
+   */
+  readonly api: IHttpApi;
+  /**
+   * custom domain nam efor the HttpApiMapping resource
+   */
+  readonly domainName: IDomainName;
+  /**
+   * stage for the HttpApiMapping resource
+   */
+  readonly stage: IStage;
+}
+
+/**
+ * The attributes used to import existing HttpApiMapping
+ */
+export interface HttpApiMappingAttributes {
+  /**
+   * The API mapping ID
+   */
+  readonly apiMappingId: string;
+}
+
+/**
+ * Create a new API mapping for API Gateway HTTP API endpoint.
+ * @resource AWS::ApiGatewayV2::ApiMapping
+ */
+export class HttpApiMapping extends Resource implements IHttpApiMapping {
+  /**
+   * import from API ID
+   */
+  public static fromHttpApiMappingAttributes(scope: Construct, id: string, attrs: HttpApiMappingAttributes): IHttpApiMapping {
```

Comment:
I like to model the API as `fromApiMappingId()` and keep the 3rd argument as `apiMappingId: string`. I find the API cleaner.

I'm ok if you want to keep what you have though.